src/main/java/uk/gov/pay/connector/gateway/stripe/handler/StripeCaptureHandler.java

```
@@ -93,17 +84,68 @@ public CaptureResponse capture(CaptureGatewayRequest request) {
+
+        LOGGER.info("Making request to Stripe to capture charge");
+        StripeCharge capturedCharge = captureWithPaymentIntentAPI(request);
+        return transferToConnectAccount(request, capturedCharge, false);
+    }
+
+    private StripeCharge queryStripeCharge(CaptureGatewayRequest request) throws GatewayException.GenericGatewayException, GatewayException.GatewayConnectionTimeoutException, GatewayErrorException {
+        var getPaymentIntentRequest = new StripeGetPaymentIntentRequest(request.getGatewayAccount(), stripeGatewayConfig, request.getGatewayTransactionId(), true);
+        String rawResponse = client.getRequestFor(getPaymentIntentRequest).getEntity();
+        StripePaymentIntent paymentIntent = jsonObjectMapper.getObject(rawResponse, StripePaymentIntent.class);
+
+        return getStripeChargeFromPaymentIntent(paymentIntent);
+    }
+
+    private StripeCharge getStripeChargeFromPaymentIntent(StripePaymentIntent paymentIntent) throws GatewayErrorException {
+        List<StripeCharge> stripeCharges = paymentIntent.getChargesCollection().getCharges();
 
         if (stripeCharges.size() != 1) {
             throw new GatewayErrorException(
-                    String.format("Expected exactly one charge associated with payment intent %s, got %s", request.getTransactionId(), stripeCharges.size()));
+                    String.format("Expected exactly one charge associated with payment intent %s, got %s", paymentIntent.getId(), stripeCharges.size()));
         }
+        return stripeCharges.get(0);
+    }
 
-        StripeCharge stripeCharge = stripeCharges.get(0);
+    private List<Fee> transferToConnectAccount(CaptureGatewayRequest request, StripeCharge capturedCharge, boolean checkForExistingTransfer) 
+            throws GatewayException.GenericGatewayException, GatewayErrorException, GatewayException.GatewayConnectionTimeoutException {
+        
+        List<Fee> feeList = generateFeeList(request, capturedCharge.getFee().orElse(0L));
```

We should probably throw an error rather than returning 0 if there is no fee present on the Stripe charge.
We're now getting the fee from the request to get the payment intent when we retry if the payment has already been captured. For this request we need to make sure we expand the balance_transactions - so logging an error would let us know if this broke somehow.
Thoughts?

---

Throwing error sounds good instead of setting the fee to `0`. `Fee` should always be present even when the payment amount is too low (1p).